### PR TITLE
Update SnapRefreshRPC and add SnapRefreshStatus RPC

### PR DESCRIPTION
**Pull request by berkayoz**
Update SnapRefreshRPC and add SnapRefreshStatus RPC

### CODE REVIEW

File: api/v1/rpc_snap_refresh_status.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+	ChangeID string `json:"changeId"`
+}
+
+// SnapRefreshStatusResponse is the response message for the SnapRefreshStatus RPC.
+type SnapRefreshStatusResponse struct {
+	// Status is the status of the snap refresh/install operation.
+	Status string `json:"status"`
+	// Completed is a boolean indicating if the snap refresh/install operation has completed.
+	// The status should be considered final when this is true.
+	Completed bool `json:"completed"`
```
Comment (bschimke95) [inline api/v1/rpc_snap_refresh_status.go:18]: Why do we have an extra field for that? Can this not be part of Status?

Comment (berkayoz) [reply api/v1/rpc_snap_refresh_status.go:18]: This is a mapping to the `Ready` field from the [snapd api](https://github.com/canonical/snapd/blob/a65eed3e1c1ab3b838ebf4aafb10741f894f6277/client/change.go#L31-L44). The status field represents the state of the operation currently, the `Ready` flag is marked as true once everything is complete (rolling back the snap on failure etc.).

Comment (bschimke95) [reply api/v1/rpc_snap_refresh_status.go:18]: Makes sense, thanks!

**Review by bschimke95 — APPROVED** [review]
LGTM

**Pull request merged**